frontend/ui/widget/dictquicklookup.lua

```diff
@@ -185,25 +254,36 @@ function DictQuickLookup:update()
             },
             {
                 {
-                    text = _("Wikipedia"),
-                    enabled = not self.wiki,
+                    text = self.wiki and _("Full page") or _("Wikipedia"),
                     callback = function()
                         UIManager:scheduleIn(0.1, function()
-                            self:lookupWikipedia()
+                            if self.wiki then
+                                self:lookupWikipedia(self.lookupword) -- get fullpage of lookupword
+                            else
+                                self:lookupWikipedia()
+                            end
                         end)
                     end,
                 },
+                -- Rotate thru available wikipedia languages
+                -- (replace previous unimplemented "Add Note")
                 {
-                    text = _("Add Note"),
-                    enabled = false,
+                    text = self.wiki and ( #self.wiki_languages > 1 and self.wiki_languages[1].." > "..self.wiki_languages[2] or self.wiki_languages[1] ) or "-",
```

Comment:
again, better not to abuse `foo and x or y` here for complex conditions like this.